### DISCUSSION TITLE
vCenter 7.03 ESXi 6.7 u1 - content library fails import

### DISCUSSION THREAD

**Discussion by MT-Homelabs** [vmware/packer-examples-for-vsphere#643] [Help] [answered]
I got everything to work expect for the part where the VM is turned off and imported into the content library. It starts the import, but hangs at 66% then eventually fails. I even tried making a brand new content library just for this test. 

Is this an issue with the versions I'm running? I saw a previous question answered regarding issues with content libraries and 6.7, wondering if this is related. 

Also, how would I alter config/common.pkrvars.hcl to remove the content library sections? 

Thanks in advance!

**Comment by virtualistic** [comment]
Dunno about 6.7 but I think you skip the content library part by editing the common.pkrvars.hcl in your config directory and set:

`common_content_library_skip_export = true`

**Reply by tenthirtyam** [reply, marked as answer]
This should work:

```hcl
// Template and Content Library Settings
common_template_conversion         = false
common_content_library_name        = null
common_content_library_ovf         = false
common_content_library_destroy     = false
common_content_library_skip_export = true
```

Setting the `common_content_library_name = null` would be sufficient however based on the logic in the HCL.

```hcl
dynamic "content_library_destination" {
    for_each = var.common_content_library_name != null ? [1] : []
    content {
      library     = var.common_content_library_name
      description = local.build_description
      ovf         = var.common_content_library_ovf
      destroy     = var.common_content_library_destroy
      skip_import = var.common_content_library_skip_export
    }
  }

```

**Reply by MT-Homelabs** [reply]
Thanks that worked. Instead of using a content library, what If I just want to have it create a template, then move that template to a folder, then delete the vm?

Thanks for the prompt response!

Edit: This worked:
```
// Template and Content Library Settings
common_template_conversion         = true
common_content_library_name        = "example"
common_content_library_ovf         = true
common_content_library_destroy     = true
common_content_library_skip_export = true

// OVF Export Settings
common_ovf_export_enabled   = true
common_ovf_export_overwrite = true
```

**Comment by tenthirtyam** [comment]
Regarding use of 6.7 this has been out of scope of the project as all 6.x versions are end of general support as well as a prior issue that was resolved in 7.0.2.

Regarding disabling the use of the content library option, see https://github.com/vmware-samples/packer-examples-for-vsphere/discussions/643#discussioncomment-6679538.

**Comment by gnoffer** [comment]
it seems that i still have exactly the same issue with vsphere 7.0.3.
as i want to run post-processor with amazon-import as AMI it is required for me that it will be exported as ova.
any tips on how to solve that ?

is there any descripton of what what "common_content_library_*" setting will result in?

**Reply by tenthirtyam** [reply]
Not specifically., but I'd be willing to add some tips [here](https://vmware-samples.github.io/packer-examples-for-vsphere/getting-started/configuration/#common).

**Reply by tenthirtyam** [reply]
But if you want it exported just to an OVA for a post-process then best to have it picked up from the export of the ovf files, followed by creating the ova archive from those files. It's ot included in this project however I do have an enhancement planned for the plugin to allow this.

**Comment by gnoffer** [comment]
thanks - that would be great. what about the timeframe for that?
actually i get a ovf exported, but need ova? what i am doing wrong ?

```
// Template and Content Library Settings
common_template_conversion         = true
common_content_library_name        = "example"
common_content_library_ovf         = true
common_content_library_destroy     = true
common_content_library_skip_export = true


// OVF Export Settings
common_ovf_export_enabled   = true
common_ovf_export_overwrite = true
```
